sycl/test-e2e/Matrix/joint_matrix_down_convert.cpp

```diff
@@ -0,0 +1,113 @@
+//==-------- joint_matrix_down_convert.cpp  - DPC++ joint_matrix------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: matrix
+
+// RUN: %{build} -o %t.out -DSYCL_EXT_ONEAPI_MATRIX_VERSION=4
+// RUN: %{run} %t.out
+
+#include "common.hpp"
+#include <iostream>
+#include <random>
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental::matrix;
+using bfloat16 = sycl::ext::oneapi::bfloat16;
+
+#define SG_SZ 16
+
+#define TM 8
+#define TN SG_SZ
```

Comment:
To avoid errors if sub group size is different.
```suggestion
#define TN 16
```

Reply:
http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-macros2